internal/helpers/secrets.go

```diff
@@ -112,7 +240,13 @@ func SyncSecret(ctx context.Context, client ctrlclient.Client, obj ctrlclient.Ob
 		// syncable-secret's Status, but...
 		dest.Data = data
 		logger.V(consts.LogLevelDebug).Info("Updating secret")
-		return client.Update(ctx, &dest)
+		if err := client.Update(ctx, &dest); err != nil {
+			return err
```

Comment:
Nit: Does it make sense to add some logging here around SyncSecret returning error? I looked at the places where `SyncSecret()` is called and it looks like the error gets returned directly to Reconcile, so it's not entirely lost. 🤷🏽

Reply:
Yeah, I think it might be okay to let the caller log it. We can add it later if it becomes an issue.